larpandoracontent/LArTwoDReco/LArClusterAssociation/CrossGapsExtensionAlgorithm.cc

```diff
@@ -51,72 +51,90 @@ void CrossGapsExtensionAlgorithm::GetListOfCleanClusters(const ClusterList *cons
 
 void CrossGapsExtensionAlgorithm::FillClusterAssociationMatrix(const ClusterVector &clusterVector, ClusterAssociationMatrix &clusterAssociationMatrix) const
 {
-    // Build lists of pointing clusters in proximity to gaps
-    LArPointingClusterList innerPointingClusterList, outerPointingClusterList;
-    this->BuildPointingClusterList(clusterVector, innerPointingClusterList, outerPointingClusterList);
+    // Build and loop over list of pointing clusters
+    LArPointingClusterList pointingClusterList;
+    this->BuildPointingClusterList(clusterVector, pointingClusterList);
```

Comment:
Is the `pointingClusterList` sorted in a meaningful, and reproducible way?

Reply:
The output pointingClusterList (std::vector) ought to have the same ordering as the input clusterVector, which has already been sorted.

Reply:
Great, this is super important! I can't see any loops over unordered containers, but if you recall such an operation then please let me know ASAP!!!